src/python/pants/engine/internals/docker_test.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+import json
+import subprocess
+
+import pytest
+
+from pants.engine.internals.docker import DockerResolveImageRequest, DockerResolveImageResult
+from pants.engine.rules import QueryRule
+from pants.testutil.rule_runner import RuleRunner
+
+
+@pytest.fixture
+def rule_runner() -> RuleRunner:
+    return RuleRunner(rules=[QueryRule(DockerResolveImageResult, (DockerResolveImageRequest,))])
+
+
+def test_resolve_image_id(rule_runner: RuleRunner) -> None:
+    subprocess.check_call(["docker", "pull", "busybox:1"])
```

Comment:
I would consider putting this check_call after the actual rule_runner request, to ensure that the image doesn't have to exist locally before invoking this rule (unless that is the case that it has to, in which case a comment as such would be good to have on the request type, I think)

Reply:
invokes docker pull from within Pants` => uses `ImagePullCache` which makes the equivalent API calls to `docker pull`

Reply:
Gotcha. 👍🏽